File: src/main/java/com/optica/services/seo/SeoComponents.java
package com.optica.services.seo;

import java.util.Map;

import org.springframework.ui.Model;

import com.optica.domain.JobOffer;

public interface SeoComponents {

	/**
	 * SEO in search
	 * */
	void generateSeoComponents(Model model, Map<String, String> filters);

	/**
	 * SEO in detail
	 * */
	void generateSeoComponent(Model model, JobOffer jobOffer);

}

File: src/main/kotlin/com/optica/services/seo/SeoComponents.kt
package com.optica.services.seo

import org.springframework.ui.Model

import com.optica.domain.JobOffer

interface SeoComponents {

    /**
     * SEO in search
     */
    fun generateSeoComponents(model: Model, filters: Map<String, String>)

    /**
     * SEO in detail
     */
    fun generateSeoComponent(model: Model, jobOffer: JobOffer)

}
